src/test_workflow/smoke_test/smoke_tests_spec/default/k-NN.yml

```diff
@@ -0,0 +1,91 @@
+---
+info:
```

Comment:
can I know some details from where we identified these tests? These tests will need modification as they are testing the feature. @reta what would be a process to fix these tests?

Reply:
We would like your inputs for this test as well. We want to test the essential APIs that won't cause any conflicts on the cluster.

Reply:
> @reta what would be a process to fix these tests?

@navneet1v I am sorry but I don't know where these tests came from in the first place.

Reply:
Yeah I added the tests from both that file and the documentation from https://opensearch.org/docs/latest/search-plugins/knn/api/ here as well.
I think for k-NN plugin, I didn't test too much as using k-NN plugin to train the models will somehow stop the cluster that blocks following tests. 
I'm willing to get some inputs/helps from k-NN maintainers regarding of how we could contribute here.

Reply:
Hey @navneet1v Do you have any other suggestions on these smoke test cases for k-NN plugin? We tried to make it light weighted so I was hesitate about adding a model training. So far I run the API for get stats. Please let me know if you think there is something else I can add to this onboarding step. Thanks!